tests/UnifiedSpecTests/EventObserver.php

```diff
@@ -270,6 +291,30 @@ private function handleEvent($event): void
             return;
         }
 
+        if (!$this->observeSensitiveCommands && $this->isSensistiveCommand($event)) {
+            return;
+        }
+
         $this->actualEvents[] = $event;
     }
+
+    /** @param CommandStartedEvent|CommandSucceededEvent|CommandFailedEvent $event */
+    private function isSensistiveCommand($event): bool
+    {
+        if (isset(self::$sensitiveCommands[$event->getCommandName()])) {
+            return true;
+        }
+
+        /* If the command or reply included a speculativeAuthenticate field,
+         * libmongoc will already have redacted it (CDRIVER-4000). Therefore, we
+         * can infer that the command was sensitive if its command or reply is
+         * empty. */
+        if (isset(self::$sensitiveCommandsWithSpeculativeAuthenticate[$event->getCommandName()])) {
+            $commandOrReply = $event instanceof CommandStartedEvent ? $event->getCommand() : $event->getReply();
+
+            return ((array) $commandOrReply === []);
```

Comment:
Interesting code sniff that feels responsible for this. I believe it is complaining about the parentheses:
```suggestion
            return (array) $commandOrReply === [];
```